docs/tsp/typeServerProtocol.ts

```diff
@@ -241,36 +269,35 @@ export namespace TypeServerProtocol {
         // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
         // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
         // only usable for the snapshot they were returned with.
+        //
+        // Snapshots are not meant to survive any changes that would make the type server throw away its internal cache. They are merely an
+        // identifier to indicate to the client that the type server will accept requests for types from that snapshot.
         GetSnapshot = 'typeServer/getSnapshot',
         // Request to get diagnostics for a specific file.
         GetDiagnostics = 'typeServer/getDiagnostics',
         // Request to get the version of diagnostics for a specific file.
         GetDiagnosticsVersion = 'typeServer/getDiagnosticsVersion',
         // Request to get the type information for a specific node.
         GetType = 'typeServer/getType',
-        // Request to get the union types for a specific type.
+        // Request to get the collection of subtypes that make up a union type.
         GetUnionTypes = 'typeServer/getUnionTypes',
-        // Request to find a member of a type.
-        GetTypeMember = 'typeServer/getTypeMember',
-        // Request to get all overloads of a function or method.
+        // Request to find an attribute of a class.
+        GetTypeAttribute = 'typeServer/getTypeAttribute',
+        // Request to get all overloads of a function or method. The returned value doesn't include the implementation signature.
         GetOverloads = 'typeServer/getOverloads',
-        // Request to get the effective meta class of a type.
-        GetEffectiveMetaClass = 'typeServer/getEffectiveMetaClass',
+        // Request to get the meta class of a type.
+        GetMetaclass = 'typeServer/getMetaclass',
         // Request to get the type of a declaration.
         GetTypeOfDeclaration = 'typeServer/getTypeOfDeclaration',
         // Request to get symbol declaration information for a node.
         GetSymbolDeclarationInfo = 'typeServer/getSymbolDeclarationInfo',
-        // Request to get the bound magic method for a type.
-        GetBoundMagicMethod = 'typeServer/getBoundMagicMethod',
-        // Request to resolve an alias declaration.
-        // This request is sent by the client to resolve an alias declaration, which may involve looking up the
-        // declaration in the type server's cache or performing additional resolution logic.
-        ResolveAliasDeclaration = 'typeServer/resolveAliasDeclaration',
+        // Request to resolve an import declaration. Example: `from module import something`. The `something` is the import declaration.
+        ResolveImportDeclaration = 'typeServer/resolveImportDeclaration',
         // Request to resolve an import.
         ResolveImport = 'typeServer/resolveImport',
```

Comment:
I don't understand this one

Reply:
ResolveImport? That is used for finding the source file for where the type server thinks an import is being pulled from. 

Example:

```
import os
```

ResolveImport on 'os' would return the on disk path to where the os.py file is.

Reply:
for us, something closer to LSP like resolveDefinition would be better since it's already implemented: can we ask for the definition of imports themselves?

Reply:
Not sure what you mean by the definition of the import? Do you mean goto def?
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_definition

That's one of the use cases for this message internally. We also use it to determine if a module is 3rd party code or not.

Reply:
We also use it on the language server side to build up source file trees which determines which files we search through for stuff like find all references.